frontend/src/components/fileUpload.js

```diff
@@ -13,20 +14,29 @@ export default function FileUpload() {
 
   async function handleOnChange(event) {
     const file = event.target.files[0];
-    const convertFileToString = (uploadedFile) =>
-      new Promise((resolve, reject) => {
-        const reader = new FileReader();
-        if (uploadedFile) {
-          reader.readAsText(uploadedFile);
-        }
-        reader.onload = () => {
-          resolve(reader.result);
-        };
-        reader.onerror = (error) => reject(error);
-      });
+    if (
+      !file.name.includes(".c") &&
```

Comment:
I think this should probably be some kind of `endsWith` test

Reply:
I have made the changes based on your review